pkg/ring/kvstore/etcd/etcd.go

```diff
@@ -0,0 +1,187 @@
+package etcd
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/ring/kvstore"
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/cortexproject/cortex/pkg/util/flagext"
+	"github.com/etcd-io/etcd/clientv3"
+	"github.com/go-kit/kit/log/level"
+)
+
+// Config for a new etcd.Client.
+type Config struct {
+	Endpoints   []string      `yaml:"endpoints"`
```

Comment:
hmm okay, `Endpoints` is each etcd leader node I guess?

Reply:
I think you can point it at _any_ etcd node.